src/components/TableRecordings.vue

```diff
@@ -103,7 +113,61 @@ export default {
         },
       ]
     };
-  }
+  },
+  methods: {
+    parseTags: function (tags) {
+      // Build a collection of tagItems - one per animal
+      const tagItems = {};
+      for (const tag of tags) {
+        const animal = tag.animal === null ? 'F/P' : tag.animal;
+        if (!tagItems[animal]) {
+          // Animal has not been seen yet
+          tagItems[animal] = {};
+        }
+        if (tag.automatic) {
+          tagItems[animal].automatic = true;
+        } else {
+          tagItems[animal].human = true;
+        }
+      }
+      // Use automatic and human status to create an ordered array of objects
+      // suitable for parsing into coloured spans
+      const result = [];
+      for (const animal of Object.keys(tagItems)) {
+        const tagItem = tagItems[animal];
+        if (tagItem.automatic && tagItem.human) {
+          result.push({
+            text: animal,
+            class: 'text-success',
+            order: 0
+          });
+        } else if (tagItem.human) {
+          result.push({
+            text: animal,
+            class: '',
+            order: 1
+          });
+        } else if (tagItem.automatic) {
+          result.push({
+            text: animal,
+            class: 'text-danger',
+            order: 2
+          });
+        }
+      }
+      // Sort the result array
+      result.sort((a,b) => {
+        return a.order - b.order;
+      });
+      // Add commas to the strings if appropriate
+      result.forEach((item, index, array) => {
+        if (index !== array.length - 1) {
+          array[index].text = item.text + ',';
+        }
+      });
```

Comment:
@mjs I did consider something like that, but it is slightly more complex than that example because it is using a method rather than a static data array. ie to do it in the template would require running collateTags(tags) again so that its length can be determined. I can't use a computed property either because it per row in the table ie slot-scope="{item: {tags}}"

So I could make it run collateTags twice, or leave it as it is, or perhaps shift the collateTag function further up the component tree ie TableRecordings.vue is called in RecordingsView.vue, where it could be part of the getRecordings() method (perhaps the best idea on reflection - but a job for another day!)

Reply:
I have now implemented this. collateTags is called by the getRecordings() methods in RecordingsView while the comma is easily handled within the template in TableRecordings.